### DISCUSSION TITLE
re-export useLinkClickHandler from react-router-dom

### DISCUSSION THREAD

**Discussion by xeoneux** [remix-run/remix#5231] [Proposals]
### What version of Remix are you using?

1.2.3

### Steps to Reproduce

There's no export of `useLinkClickHandler` for creating custom fancy pants links as per the react-router dom API doc

### Expected Behavior

It should be exported from the remix module

### Actual Behavior

It's not exported from the remix module

**Comment by gfohl** [comment]
This would be great, and even better if we can add the prefetching to it as well!

**Comment by xeoneux** [comment]
Hey remix team, any update on this? @mcansh @chaance

**Comment by sergiodxa** [comment]
> 

The prefetching works with an exported component https://remix.run/docs/en/v1/api/remix#prefetchpagelinks-, the component needs to be rendered in order to render the `<link rel="prefetch">` tags so it can't work as a hook unless the hook return the JSX for those tags.

**Comment by sergiodxa** [comment]
@xeoneux You could still import the hook from `react-router-dom` even if it's not exported from `@remix-run/react`

```ts
import { useLinkClickHandler } from "react-router-dom";
```

This should work.

**Comment by machour** [comment]
Converting this to a Proposal discussion, as per our [Development Process](https://remix.run/docs/en/v1/pages/contributing#development-process).
